### DISCUSSION TITLE
Tangent/Local axis of a Polygon In Mesh Object input

### DISCUSSION THREAD

**Discussion by Gerstmann-Bradley** [JacquesLucke/animation_nodes#1721] [Q&A] [answered]
In "Mesh Object input" node, we can get polygon center and polygon normal.
But with a single normal we don't know the local axis of each polygon.
Basically the goal is to align the red edge parallel to blue edge in all rotation of our object.
I think there should be a way, since in Polygon Offset node we can manipulate "Local Axis",
but I don't know how to do so with Mesh Object Input node.
![image](https://user-images.githubusercontent.com/55994236/117797620-37364a80-b21e-11eb-9547-7e6ccf021e35.png)

**Comment by Gerstmann-Bradley** [comment, marked as answer]
The solution:
Disable use the World Space. and transform the matrix with mother objects' transformation.
![image](https://user-images.githubusercontent.com/55994236/117804335-a7949a00-b225-11eb-9f99-fbca9c3ffcf0.png)

**Reply by Gerstmann-Bradley** [reply]
@OmarEmaraDev There's one additional problem though.
In the follow example, the Left is Blender Object instancing by face. the Right is AN's instance.
The difference is that Blender Object instancing by face has object locked onto a specific edge of polygon.
Do you know deal with such kind of rotation in AN using Object Mesh data Node?
![image](https://user-images.githubusercontent.com/55994236/118407119-33a32900-b64d-11eb-8d0b-ca40f6c2bf51.png)

**Reply by OmarEmaraDev** [reply]
@Gerstmann-Bradley Use the *Extract Polygon Transforms* node in those cases.

**Reply by Gerstmann-Bradley** [reply]
> @Gerstmann-Bradley Use the _Extract Polygon Transforms_ node in those cases.

I see. Thanks!
